vignettes/forcats.Rmd

```diff
@@ -44,17 +43,59 @@ To do so, we can use the function `fct_infreq()`:
 
 ```{r fct-infreq-hair}
 #| fig.alt: >
-#|   The A bar chart of hair color, now ordered so that the least
+#|   The bar chart of hair color, now ordered so that the least
 #|   frequent colours come first and the most frequent colors come last.
 #|   This makes it easy to see that the most common hair color is none 
-#|   (~35), followed by brown (~18), then black (~12).
-ggplot(starwars, aes(x = fct_infreq(hair_color))) + 
-  geom_bar() + 
-  coord_flip()
+#|   (~35), followed by brown (~18), then black (~12). Surprisingly,
+#|   NAs are at the top of the graph, even though there are ~5 NAs and 
+#|   other colors have smaller values.
+ggplot(starwars, aes(y = fct_infreq(hair_color))) + 
+  geom_bar()
 ```
 
 Note that `fct_infreq()` it automatically puts NA at the top, even though that doesn't have the smallest number of entries.
 
+Its's a little surprising that the `NA` bar isn't ordered by frequency.
+To understand why, we need to make a brief digression to discuss `NA`s in values vs `NA`s in levels
+
+## `NAs`s in levels and values
+
+There are two ways to represent a missing value in a factor:
+
+-   You can include it in the values of the factor; it does not appear in the levels and `is.na()` reports it as missing. This is the usual way to encode a missing value in a factor:
+
+```{r}
+f <- factor(c("x", "y", NA))
+levels(f)
+is.na(f)
+```
+
+-   You can include in the levels of the factor and `is.na()` does not report it as missing. This requires a little more work to create:
+
+```{r}
+f <- factor(c("x", "y", NA), exclude = NULL)
+levels(f)
+is.na(f)
+```
+
+`NA`s in the values tend to be best for data analysis, since `is.na()` works as you'd expect.
+`NA`s in the levels are often better for display since you can control where they appear.
```

Comment:
I would move these two sentences to the end of this section since they're like a summary.

Also, I wonder about saying something like 

> You should be careful about using `fct_na_value_to_level()` in a data transformation pipeline where you overwrite your data since that will affect all downstream analysis as well. This should only be done if you will be using that updated data frame or creating displays or summaries, chances are you don't want NAs as an explicit level in your data analysis. If you do, you might consider relabeling them as something like "Unknown" or "Not available".

Maybe this is an overreach, but I feel like this function should be used as a quick fix, but if you really care about those `NA`s it's more practical to not make them `NA`s but something else, so they don't accidentally get swooped up by another function that treats NAs in a special way, say in a modelling context or something...

Reply:
> If you do, you might consider relabeling them as something like "Unknown" or "Not available".

Which you'd do with `fct_na_value_to_level(x, name = "Unknown")` if we add the idea of `fct_explicit_na(na_level =)` back in